index.js

```diff
@@ -49,16 +69,71 @@ client.on(Events.InteractionCreate, interaction => {
         interaction.reply('Pong!');
     }
 
-    // inversion slash command
+    // Inversion slash command
     if (interaction.commandName === 'inversion') {
         const message = interaction.options.get('message').value;
         
         // Reverse string
-        let invertedMessage = message.split("").reverse().join("");
+        let invertedMessage = message.split('').reverse().join('');
 
         interaction.reply(invertedMessage);
     }
 })
 
+// Listen for threads being created
+client.on(Events.ThreadCreate, async (thread) => {
+    console.log(`Thread created: ${thread.name} in ${thread.parent.name}`);
+     
+    try {
+        // Grab message thread was created
+        const threadMessage = await thread.fetchStarterMessage();
+        console.log(`Thread created from message: ${threadMessage.content}`);
+
+        // Automate thread title by grabbing from starter message 
+        const threadTitleMatch = threadMessage.content.match(THREAD_TITLE_REGEX);
+        if (threadTitleMatch && threadTitleMatch[0]) {
+            thread.setName(threadTitleMatch[0]);
+        }
+    } 
+    catch (error) {
+        console.error('Error fetching starter message:', error);
+    }
+});
+
+// Listen for threads being archived
+client.on(Events.ThreadUpdate, (oldThread, newThread) => {
+    if (oldThread.archived === false && newThread.archived === true) {
```

Comment:
Pro tip: skip the `===` comparison. You can skip the `=== true` part, `newThread.archived` already evaluates to true, so you can use it directly (or `!` to check if it's false). Makes the code a little cleaner. Your logic was on point though! 👍